src/network/networkctl.c

```diff
@@ -657,84 +678,116 @@ static int next_lldp_neighbor(FILE *f, sd_lldp_neighbor **ret) {
 
-                printf("%*s",
-                       (int) strlen(prefix),
-                       c == 0 ? prefix : "");
+                r = table_add_cell(table, NULL, TABLE_EMPTY, NULL);
+                if (r < 0)
+                        return r;
+
+                r = table_add_cell_full(table, NULL, TABLE_STRING, c == 0 ? prefix : "", (size_t) -1, (size_t) -1, 0, 100, 0);
+                if (r < 0)
+                        return r;
 
                 (void) sd_lldp_neighbor_get_system_name(n, &system_name);
                 (void) sd_lldp_neighbor_get_port_id_as_string(n, &port_id);
                 (void) sd_lldp_neighbor_get_port_description(n, &port_description);
 
-                printf("%s on port %s", strna(system_name), strna(port_id));
-
-                if (!isempty(port_description))
-                        printf(" (%s)", port_description);
+                if (asprintf(&str, "%s on port %s%s%s%s",
+                             strna(system_name), strna(port_id),
+                             isempty(port_description) ? "" : " (",
+                             port_description,
+                             isempty(port_description) ? "" : ")") < 0)
+                        return -ENOMEM;
 
-                putchar('\n');
+                r = table_add_cell(table, NULL, TABLE_STRING, str);
+                if (r < 0)
+                        return r;
 
                 c++;
         }
 
         return c;
 }
 
-static void dump_ifindexes(const char *prefix, const int *ifindexes) {
+static int dump_ifindexes(Table *table, const char *prefix, const int *ifindexes) {
         unsigned c;
+        int r;
 
         assert(prefix);
 
         if (!ifindexes || ifindexes[0] <= 0)
-                return;
+                return 0;
 
         for (c = 0; ifindexes[c] > 0; c++) {
                 char name[IF_NAMESIZE+1];
 
-                printf("%*s",
-                       (int) strlen(prefix),
-                       c == 0 ? prefix : "");
+                r = table_add_cell(table, NULL, TABLE_EMPTY, NULL);
+                if (r < 0)
+                        return r;
 
-                if (if_indextoname(ifindexes[c], name))
-                        fputs(name, stdout);
-                else
-                        printf("%i", ifindexes[c]);
+                r = table_add_cell_full(table, NULL, TABLE_STRING, c == 0 ? prefix : "", (size_t) -1, (size_t) -1, 0, 100, 0);
+                if (r < 0)
+                        return r;
 
-                fputc('\n', stdout);
+                if (if_indextoname(ifindexes[c], name)) {
+                        r = table_add_cell(table, NULL, TABLE_STRING, name);
+                        if (r < 0)
+                                return r;
+                } else {
+                        r = table_add_string_cell(table, NULL, "%i", ifindexes[c]);
+                        if (r < 0)
+                                return r;
+                }
```

Comment:
hmm, i wonder if it would make sense to add cell data type `TABLE_IFINDEX` that takes an `int`and on display converts the `int` value to an interface name and shows the number otherwise. i.e. convert the ifindex to a name in the table formatter code instead of here in advance. just an idea though